src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -23,6 +23,15 @@ namespace Microsoft.Quantum.QsCompiler.CompilationBuilder.DataStructures
     /// </summary>
     internal class CodeLine
     {
+        internal enum StringContext
```

Comment:
Suggest to make a comment about how these states relate to context in the text:
NOS " OS
NOS $" OIS { OIA " OSIOIA

Reply:
I've added an example to the doc comment for the enum.